src/math.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+if dependencies::has_bc; then
+  # bc is better than awk because bc has no integer limits.
+  function math::calculate() {
+    echo "$*" | bc
+  }
+elif dependencies::has_awk; then
+  function math::calculate() {
+    awk "BEGIN { print ""$*"" }"
+  }
+fi
```

Comment:
this reminds me of C when declaring different function implementations depending on the OS. I kind of like it, and I believe it's performance-optimized, nice 👍🏼